presto-hive/src/main/java/com/facebook/presto/hive/HiveZeroRowFileCreator.java

```diff
@@ -71,6 +71,9 @@ public HiveZeroRowFileCreator(
     @Override
     public void createFiles(ConnectorSession session, HdfsContext hdfsContext, Path destinationDirectory, List<String> fileNames, StorageFormat format, HiveCompressionCodec compressionCodec, Properties schema)
     {
+        if (fileNames.isEmpty()) {
```

Comment:
It's not clear to me why we need this, because it will be used for a for-each loop. Do we add this because we don't want to go through `generateZeroRowFile`?

Reply:
Yes, avoid unneeded `generateZeroRowFile`